### DISCUSSION TITLE
Latex representation of derived units with fractions

### DISCUSSION THREAD

**Discussion by dvd101x** [josdejong/mathjs#3168] [Ideas]
In version 12.4.0 when getting the latex representation of a derived unit with fractions like
```js
math.parse("10 km/hr").toTex()    // yields "\\frac{10~\\mathrm{km}}{\\mathrm{hr}}"
math.parse("400 W m/K").toTex()   // yields "\\frac{400~\\mathrm{W}~\\mathrm{m}}{\\mathrm{K}}"
```
We get:

$$\frac{10~\mathrm{km}}{\mathrm{hr}} ,\ \frac{400~\mathrm{W}~\mathrm{m}}{\mathrm{K}}$$

Maybe it should be:
$$10~\frac{\mathrm{km}}{\mathrm{hr}}  , \ 400~\frac{\mathrm{W}~\mathrm{m}}{\mathrm{K}}$$

**Comment by josdejong** [comment]
Hm, that is an interesting idea. I'm not sure how easy it is to implement. It would need to recognize a pattern of a numeric value followed by units (which can also be something like `m^3/h`.

**Reply by dvd101x** [reply]
I'm not very familiar with the parsing algorithm, so I'm also not sure if it would be easy.

For my specific need I'm planning to use parse twice, once to parse the value and once for the unit, which I think will accomplish my purpose.

Something similar to this:

```js
function unitToTex(myUnit) {
  const uj = myUnit.toJSON();
  return (
    math.parse(uj.value.toString()).toTex() +
    "\\ " +
    math.parse(uj.unit).toTex()
  );
}
```

Still to be defined is this behavior of parsing units to tex is desirable in the library.
